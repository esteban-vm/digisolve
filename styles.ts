import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
