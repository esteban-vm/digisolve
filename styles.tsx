import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
