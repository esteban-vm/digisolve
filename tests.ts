import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
