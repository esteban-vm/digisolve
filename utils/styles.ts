import { css } from '@emotion/react'

export enum breakPoints {
  /** Big tablets to smaller laptop screens */
  lg = '1200px',
  /** Smaller tablets to larger tablet screens */
  md = '1023px',
  /** Larger phones to smaller tablet screens */
  sm = '767px',
  /** Smaller to larger mobile phones */
  xs = '479px',
}

export const mediaQuery = (breakPoint: keyof typeof breakPoints) => {
  return `@media only screen and (max-width: ${breakPoints[breakPoint]})`
}

export const globalStyles = css`
  @import url('/css/fluid.min.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;
    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;
    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;
    --color-dark: #888;
    --color-darker: #333;
    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;
    --color-shadow: #efefefef;
    --color-facebook: #3b5998;
    --color-twitter: #00aced;
    --color-google-plus: #dd4b39;
    --color-youtube: #f00;
    --color-instagram: #517fa4;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;

    ${mediaQuery('lg')} {
      font-size: 14px;
    }

    ${mediaQuery('md')} {
      font-size: 12px;
    }

    ${mediaQuery('sm')} {
      font-size: 10px;
    }
  }

  * {
    :focus {
      outline: none;
    }

    ::selection {
      background-color: var(--color-primary);
      color: var(--color-light);
    }

    ::-webkit-scrollbar {
      width: 0.5rem;

      &-track {
        box-shadow: inset 0 0 6px var(--color-darker);
      }

      &-thumb {
        background-color: var(--color-primary);
        outline: 1px solid var(--color-dark-primary);
        border-radius: 10px;
      }
    }
  }

  img,
  button,
  a {
    user-select: none;
  }

  ul {
    list-style: none;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;

    ${mediaQuery('md')} {
      padding: 0 2%;
    }
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }

  .section {
    &__heading {
      font-size: 250%;
      text-transform: capitalize;
      color: var(--color-black);
      margin-bottom: 20px;
      margin-top: 150px;

      ${mediaQuery('lg')} {
        margin-top: 90px;
      }

      ${mediaQuery('sm')} {
        margin-top: 60px;
        margin-bottom: 10px;
      }
    }

    &__subheading {
      font-size: 150%;
      color: var(--color-grey);
      font-weight: 300;
      margin-bottom: 30px;
      line-height: 1.5;

      ::after {
        content: ' ';
        display: block;
        height: 2px;
        width: 100px;
        background-color: var(--color-primary);
        margin: 30px auto 0;

        ${mediaQuery('sm')} {
          margin-top: 15px;
        }
      }
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
