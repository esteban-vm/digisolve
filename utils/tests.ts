import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
