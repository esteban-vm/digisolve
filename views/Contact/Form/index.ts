export { default, type FormFields } from './Form'
