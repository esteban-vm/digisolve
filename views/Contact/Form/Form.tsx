import type { Component } from '@/types'
import { useState } from 'react'
import { useForm, type SubmitHandler } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import { string, boolean, object, type ObjectSchema } from 'yup'
import { Button, Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'
import Toast from '../Toast'

type FormValues = {
  name: string
  email: string
  newsletter: boolean
  message: string
}

const validationSchema: ObjectSchema<FormValues> = object().shape({
  name: string().trim().required(),
  email: string().trim().email().required(),
  newsletter: boolean().default(true),
  message: string().trim().required(),
})

const defaultValues: FormValues = {
  name: '',
  email: '',
  newsletter: false,
  message: '',
}

const FormComponent: Component = (props) => {
  const [formValues, setFormValues] = useState(defaultValues)

  const {
    register,
    handleSubmit,
    reset,
    formState: { isSubmitSuccessful },
  } = useForm({ resolver: yupResolver(validationSchema), defaultValues })

  const onSubmit: SubmitHandler<FormValues> = (data) => {
    setFormValues({ ...formValues, ...data })
    reset()
  }

  return (
    <>
      <form aria-label='Send a message' onSubmit={handleSubmit(onSubmit)} noValidate {...props}>
        <Grid.Row>
          <Grid.Col isOneThird>
            <label htmlFor='name'>Name</label>
          </Grid.Col>
          <Grid.Col isTwoThirds>
            <input
              type='text'
              id='name'
              placeholder='your name'
              spellCheck={false}
              autoCorrect='off'
              className='field'
              {...register('name')}
            />
          </Grid.Col>
        </Grid.Row>
        <Grid.Row>
          <Grid.Col isOneThird>
            <label htmlFor='email'>Email</label>
          </Grid.Col>
          <Grid.Col isTwoThirds>
            <input type='email' id='email' placeholder='your email' className='field' {...register('email')} />
          </Grid.Col>
        </Grid.Row>
        <Grid.Row>
          <Grid.Col isOneThird>
            <label htmlFor='newsletter'>Newsletter?</label>
          </Grid.Col>
          <Grid.Col isTwoThirds>
            <input type='checkbox' id='newsletter' className='check' defaultChecked {...register('newsletter')} />
            Yes, Please
          </Grid.Col>
        </Grid.Row>
        <Grid.Row>
          <Grid.Col isOneThird>
            <label htmlFor='message'>Drop us a line</label>
          </Grid.Col>
          <Grid.Col isTwoThirds>
            <textarea id='message' placeholder='your message' className='field' {...register('message')} />
          </Grid.Col>
        </Grid.Row>
        <Grid.Row>
          <Grid.Col isOneThird />
          <Grid.Col isTwoThirds>
            <Button text='Send it!' isSubmit />
          </Grid.Col>
        </Grid.Row>
        <Toast isOpen={isSubmitSuccessful}>
          <p>{formValues.name}</p>
          <p>{formValues.email}</p>
          <p>{formValues.newsletter ? 'yes' : 'no'}</p>
          <p>{formValues.message}</p>
        </Toast>
      </form>
    </>
  )
}

const Form = styled(FormComponent)`
  text-align: left;
  width: 60%;
  margin: 0 auto;
  color: var(--color-grey);

  ${mediaQuery('sm')} {
    width: 70%;
  }

  ${mediaQuery('xs')} {
    width: 90%;
  }

  label {
    user-select: none;

    ${mediaQuery('xs')} {
      margin-top: 5px;

      ::after {
        content: ':';
      }
    }
  }

  .field {
    width: 95%;
    padding: 8px;
    border-radius: 3px;
    border: 1px solid #ccc;
    background-color: var(--color-white);

    ::placeholder {
      text-transform: capitalize;
      opacity: 0.7;
    }

    :is(textarea) {
      height: 100px;
      resize: none;
    }
  }

  .check {
    margin: 10px 5px 10px 0;

    ${mediaQuery('xs')} {
      margin-top: 2px;
    }
  }
`

export default Form
