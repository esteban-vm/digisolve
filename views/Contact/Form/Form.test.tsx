import { render, cleanup, screen, create, userEvent } from '@/tests'
import { mediaQuery } from '@/styles'
import Form from './Form'

describe('ðŸ§ª FORM:', () => {
  describe('display tests:', () => {
    beforeEach(() => {
      render(<Form />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const form = screen.getByRole('form', { name: /^send a message$/i })
      expect(form).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the field for name', () => {
        const nameField = screen.getByPlaceholderText(/^your name$/i)
        expect(nameField).toBeInTheDocument()
        expect(nameField).toBeVisible()
        expect(nameField).toHaveAccessibleName(/^name$/i)
      })

      it('the field for email', () => {
        const emailField = screen.getByPlaceholderText(/^your email$/i)
        expect(emailField).toBeInTheDocument()
        expect(emailField).toBeVisible()
        expect(emailField).toHaveAccessibleName(/^email$/i)
      })

      it('the checkbox for newsletter', () => {
        const checkbox = screen.getByRole('checkbox', { name: /^newsletter\?$/i })
        expect(checkbox).toBeInTheDocument()
        expect(checkbox).toBeVisible()
      })

      it('the text box for message', () => {
        const messageField = screen.getByPlaceholderText(/^your message$/i)
        expect(messageField).toBeInTheDocument()
        expect(messageField).toBeVisible()
        expect(messageField).toHaveAccessibleName(/^drop us a line$/i)
      })
    })
  })

  describe('submit tests:', () => {
    beforeEach(() => {
      render(<Form />)
    })

    afterEach(cleanup)

    it('should fail with all fields empty', async () => {
      const submitButton = screen.getByRole('button', { name: /^send it\!$/i })
      await userEvent.click(submitButton)
      const notification = screen.queryByRole('alert')
      const nameField = screen.getByPlaceholderText(/^your name$/i)
      expect(notification).not.toBeInTheDocument()
      expect(nameField).toHaveFocus()
    })

    it('should succeed with all fields full', async () => {
      const nameField = screen.getByPlaceholderText(/^your name$/i)
      const emailField = screen.getByPlaceholderText(/^your email$/i)
      const messageField = screen.getByPlaceholderText(/^your message$/i)
      const submitButton = screen.getByRole('button', { name: /^send it\!$/i })
      await userEvent.type(nameField, 'John Doe')
      await userEvent.type(emailField, 'test@example.com')
      await userEvent.type(messageField, 'test message')
      await userEvent.click(submitButton)
      const notification = await screen.findByRole('alert')
      expect(notification).toBeInTheDocument()
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Form />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    describe('should have correct styles on:', () => {
      it('extra small screen devices', () => {
        const media = mediaQuery('xs')
        expect(tree).toHaveStyleRule('width', '90%', { media })
        expect(tree).toHaveStyleRule('margin-top', '5px', { target: 'label', media })
        expect(tree).toHaveStyleRule('margin-top', '2px', { target: '.check', media })
      })

      it('small screen devices', () => {
        expect(tree).toHaveStyleRule('width', '70%', { media: mediaQuery('sm') })
      })
    })
  })
})
