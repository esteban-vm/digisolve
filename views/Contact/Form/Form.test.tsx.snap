// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª FORM: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: left;
  width: 60%;
  margin: 0 auto;
  color: var(--color-grey);
}

@media only screen and (max-width: 768px) {
  .emotion-0 {
    width: 70%;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    width: 90%;
  }
}

.emotion-0 label {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media only screen and (max-width: 480px) {
  .emotion-0 label {
    margin-top: 5px;
  }

  .emotion-0 label::after {
    content: ':';
  }
}

.emotion-0 .field {
  width: 95%;
  padding: 8px;
  border-radius: 3px;
  border: 1px solid #ccc;
  background-color: var(--color-white);
}

.emotion-0 .field::-webkit-input-placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-0 .field::-moz-placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-0 .field:-ms-input-placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-0 .field::placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-0 .field:is(textarea) {
  height: 100px;
  resize: none;
}

.emotion-0 .check {
  margin: 10px 5px 10px 0;
}

@media only screen and (max-width: 480px) {
  .emotion-0 .check {
    margin-top: 2px;
  }
}

.emotion-1 {
  cursor: pointer;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: var(--color-primary);
  color: var(--color-white);
}

.emotion-1:hover {
  background-color: var(--color-dark-primary);
}

.emotion-2 {
  text-align: center;
  color: var(--color-secondary);
  background-color: var(--color-light);
  border: 2px solid var(--color-primary);
  border-radius: 5px;
  max-width: 60%;
}

@media only screen and (max-width: 480px) {
  .emotion-2 {
    max-width: 90%;
  }
}

.emotion-2 h2 {
  margin: 0;
  color: var(--color-primary);
}

.emotion-2 .content {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 .icon {
  color: inherit;
  margin: 10px auto;
}

.emotion-2 .btn {
  margin-top: 10px;
}

.emotion-3 {
  cursor: pointer;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: transparent;
  color: var(--color-primary);
}

.emotion-3:hover {
  background-color: var(--color-primary);
  color: var(--color-white);
}

<form
  aria-label="Send a message"
  className="emotion-0"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
    </div>
    <div
      className="col span_2_of_3 "
      role="presentation"
    >
      <input
        autoCorrect="off"
        className="field"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="your name"
        spellCheck={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
    </div>
    <div
      className="col span_2_of_3 "
      role="presentation"
    >
      <input
        className="field"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="your email"
        type="email"
      />
    </div>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <label
        htmlFor="newsletter"
      >
        Newsletter?
      </label>
    </div>
    <div
      className="col span_2_of_3 "
      role="presentation"
    >
      <input
        className="check"
        id="newsletter"
        name="newsletter"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
      />
      Yes, Please
    </div>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <label
        htmlFor="message"
      >
        Drop us a line
      </label>
    </div>
    <div
      className="col span_2_of_3 "
      role="presentation"
    >
      <textarea
        className="field"
        id="message"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="your message"
      />
    </div>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    />
    <div
      className="col span_2_of_3 "
      role="presentation"
    >
      <button
        className="emotion-1"
        type="submit"
      >
        Send it!
      </button>
    </div>
  </div>
  <dialog
    className="emotion-2"
  >
    <div
      className="content"
    >
      <h2>
        Submitted successfully
      </h2>
      <svg
        aria-hidden={true}
        className="animate__animated animate__heartBeat animate__infinite icon"
        color="var(--color-primary)"
        fill="currentColor"
        height="3rem"
        role="img"
        viewBox="0 0 640 512"
        width="3rem"
      >
        <path
          d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4l217.6 163.2c11.4 8.5 27 8.5 38.4 0l57.4-43c23.9-59.8 79.7-103.3 146.3-109.8l13.9-10.4c12.1-9.1 19.2-23.3 19.2-38.4 0-26.5-21.5-48-48-48H48zm246.4 275.2a63.9 63.9 0 01-76.8 0L0 176v208c0 35.3 28.7 64 64 64h296.2c-25.1-30.4-40.2-69.5-40.2-112 0-5.6.3-11.1.8-16.6l-26.4 19.8zM640 336c0-79.5-64.5-144-144-144s-144 64.5-144 144 64.5 144 144 144 144-64.5 144-144zm-76.7-43.3c6.2 6.2 6.2 16.4 0 22.6l-72 72c-6.2 6.2-16.4 6.2-22.6 0l-40-40c-6.2-6.2-6.2-16.4 0-22.6s16.4-6.2 22.6 0l28.7 28.7 60.7-60.7c6.2-6.2 16.4-6.2 22.6 0z"
        />
      </svg>
      <div>
        <p />
        <p />
        <p>
          yes
        </p>
        <p />
      </div>
      <button
        className="btn emotion-3"
        onClick={[Function]}
        type="button"
      >
        ok
      </button>
    </div>
  </dialog>
</form>
`;
